src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -35,16 +36,13 @@ RandomKitchenSinksDotFeatures::RandomKitchenSinksDotFeatures(
 
 SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients()
 {
-	SGVector<float64_t> vec = generate_random_parameter_vector();
-	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	SGVector<float64_t> vec;
```

Comment:
you can move this variable inside the loop right?

Reply:
right, but we would be redeclaring it every loop iteration won't we?. That's why I defined it outside.

Reply:
any good compiler will see that, so it won't harm the codegen. Writing declarations like this is more of ANSI C style :D

Reply:
Thanks for everything till now ^^

Reply:
@NanuSai make sure to read a bit on C++ compiler optimizations. The code you write is not necessarily what will be executed on the machine, as gil pointed out above. If you look at the assembly of the compiled file, you will see that defining the variable before or in the loop leads to exactly the same machine code.